lib/search.rb

```diff
@@ -646,7 +646,7 @@ def post_action_type_filter(posts, post_action_type)
     end
   end
 
-  advanced_filter(/^\@([a-zA-Z0-9_\-.]+)$/i) do |posts, match|
+  advanced_filter(/^\@(\S+)$/i) do |posts, match|
     username = match.downcase
 
     user_id = User.where(staged: false).where(username_lower: username).pluck_first(:id)
```

Comment:
The username should be normalized in order to fully work with Unicode.

```suggestion
    username = User.normalize_username(match)

    user_id = User.not_staged.where(username_lower: username).pluck_first(:id)
```